app/models/search.server.ts

```diff
@@ -184,3 +173,92 @@ export function getSearchResultsFromMeanings(
     take: take,
   })
 }
+
+export interface Canadianism {
+  headword: string
+  canadianismDescription: string
+  id: number
+}
+
+// case sensitivity not working; check collation
+export function getSearchResultCanadianisms(
+  text: string,
+  skip: number = 0,
+  take: number = DEFAULT_PAGE_SIZE,
+  caseSensitive: boolean = false
+) {
+  if (text.length === 0) {
+    throw new Error(`Text ("${text}") length must be greater than zero`)
+  }
+
+  const searchWildcard = `%${text}%`
+
+  return prisma.$queryRaw<
+    Canadianism[]
+  >`SELECT det_entries.headword as headword,
+  det_meanings.canadianism_type_comment, det_meanings.id
+  FROM det_meanings, det_entries
+  WHERE det_meanings.entry_id = det_entries.id AND
+  IF(${caseSensitive}, 
+    (det_meanings.canadianism_type_comment) LIKE (${searchWildcard}), 
+    LOWER(det_meanings.canadianism_type_comment) LIKE LOWER(${searchWildcard}))  
+  ORDER BY LOWER(det_entries.headword) ASC LIMIT ${take} OFFSET ${skip}`
+}
+
+export interface UsageNote {
+  headword: string
+  partOfSpeech: string
+  usage: string
+  id: number
+}
+
+// case sensitivity not working; check collation
+export function getSearchResultUsageNotes(
+  text: string,
+  skip: number = 0,
+  take: number = DEFAULT_PAGE_SIZE,
+  caseSensitive: boolean = false
+) {
+  if (text.length === 0) {
+    throw new Error(`Text ("${text}") length must be greater than zero`)
+  }
+
+  const searchWildcard = `%${text}%`
+
+  return prisma.$queryRaw<UsageNote[]>`SELECT det_entries.headword as headword,
+  det_meanings.usage, det_meanings.partofspeech, det_meanings.id
+  FROM det_meanings, det_entries
+  WHERE det_meanings.entry_id = det_entries.id AND
+  IF(${caseSensitive}, 
+    (det_meanings.usage) LIKE (${searchWildcard}), 
+    LOWER(det_meanings.usage) LIKE LOWER(${searchWildcard}))  
+  ORDER BY LOWER(det_entries.headword) ASC LIMIT ${take} OFFSET ${skip}`
+}
+
+export interface FistNote {
+  headword: string
+  fistNote: string
+  id: number
+}
+
+// case sensitivity not working; check collation
+export function getSearchResultFistNotes(
```

Comment:
Again, similar behaviour between these `getSearchResult...` functions. I know that using the raw SQL in Prisma is finicky, so maybe that prevents easy refactoring...? Or does it?

Reply:
OK for now.